### PR TITLE
BaseHtml helper's checkbox and radio code has been abstracted as suggested in #11168

### PR DESCRIPTION
**Pull request by cesarnicola**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | no |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues | #11168 |

### CODE REVIEW

**Comment by samdark** [comment]
Looks very good to me. @yiisoft/core-developers opinions?

File: framework/helpers/BaseHtml.php [hunk 1]
```diff
@@ -1445,27 +1399,29 @@ public static function activeRadio($model, $attribute, $options = [])
...
      */
     public static function activeCheckbox($model, $attribute, $options = [])
     {
+        return static::activeBooleanInput('checkbox', $model, $attribute, $options);
+    }
+
+    /**
+     * Generates a boolean input
+     * This method is mainly called by [[activeCheckbox()]] and [[activeRadio()]].
+     * @param string $type the input type. This can be 'radio' or 'checkbox'.
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: <code>This can be either \`radio\` or \`checkbox\`</code>

File: framework/helpers/BaseHtml.php [hunk 2]
```diff
@@ -1445,27 +1399,29 @@ public static function activeRadio($model, $attribute, $options = [])
...
+     * @param Model $model the model object
+     * @param string $attribute the attribute name or expression. See [[getAttributeName()]] for the format
+     * about attribute expression.
+     * @param array $options the tag options in terms of name-value pairs.
+     * See [[booleanInput()]] for details about accepted attributes.
+     * @return string the generated input list
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: `@since 2.0.9`

File: framework/helpers/BaseHtml.php [hunk 3]
```diff
@@ -1445,27 +1399,29 @@ public static function activeRadio($model, $attribute, $options = [])
...
-     *   The checkbox will be enclosed by the label tag. Note that if you do not specify this option, a default label
-     *   will be used based on the attribute label declaration in the model. If you do not want any label, you should
-     *   explicitly set this option as null.
-     * - labelOptions: array, the HTML attributes for the label tag. This is only used when the "label" option is specified.
-     *
-     * The rest of the options will be rendered as the attributes of the resulting tag. The values will
-     * be HTML-encoded using [[encode()]]. If a value is null, the corresponding attribute will not be rendered.
-     * See [[renderTagAttributes()]] for details on how attributes are being rendered.
+     * @param array $options the tag options in terms of name-value pairs.
+     * See [[booleanInput()]] for details about accepted attributes.
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: `@see booleanInput() for details about accepted attributes`

File: framework/helpers/BaseHtml.php (hunk 2 above)
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: Add a line bellow `@see booleanInput`

File: framework/helpers/BaseHtml.php [hunk 4]
```diff
@@ -672,48 +672,35 @@ public static function textarea($name, $value = '', $options = [])
...
     }
 
     /**
      * Generates a checkbox input.
      * @param string $name the name attribute.
      * @param boolean $checked whether the checkbox should be checked.
+     * @param array $options the tag options in terms of name-value pairs.
+     * See [[booleanInput()]] for details about accepted attributes.
+     *
+     * @return string the generated checkbox tag
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: Add a line bellow `@see booleanInput`

File: framework/helpers/BaseHtml.php [hunk 5]
```diff
@@ -672,48 +672,35 @@ public static function textarea($name, $value = '', $options = [])
...
-     *   When this option is specified, the radio button will be enclosed by a label tag.
-     * - labelOptions: array, the HTML attributes for the label tag. Do not set this option unless you set the "label" option.
-     *
-     * The rest of the options will be rendered as the attributes of the resulting radio button tag. The values will
-     * be HTML-encoded using [[encode()]]. If a value is null, the corresponding attribute will not be rendered.
-     * See [[renderTagAttributes()]] for details on how attributes are being rendered.
+     * @param array $options the tag options in terms of name-value pairs.
+     * See [[booleanInput()]] for details about accepted attributes.
      *
      * @return string the generated radio button tag
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: Add a line bellow `@see booleanInput`

**Comment by SilverFire** [comment]
I've left my comments for the PHPDoc, but the code is good.
We also need a CHANGELOG line

**Comment by cesarnicola** [comment]
@SilverFire, thanks for your comments and suggestions! I think I've introduced them all but was a bit unsure about the proper placement for the @see directive.
Please check it when you have a chance and let me know if we need other tweaks.
Thanks!

**Comment by samdark** [comment]
Merged with minor adjustments https://github.com/yiisoft/yii2/commit/348cc17dbba72767a6eeee7a3b183cf9df0fce35. Thanks!

**Pull request closed without merging**